lib/Sema/TypeCheckDecl.cpp

```diff
@@ -6693,6 +6693,60 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
                                 desiredAccessScope, access);
       }
       TC.checkConformancesInContext(ED, ED);
+
+      // If we just extended a nominal type with a protocol conformance (which
+      // may have just gotten derived), members which are synthesized might need
+      // to be visited for validation (since the extended type might have
+      // otherwise finished being validated).
+      if (auto extendedTy = ED->getExtendedType()) {
+        if (auto nominal = extendedTy->getAnyNominal()) {
+          // Put together the list of protocols that the nominal type inherited
+          // from this extension.
+          SmallVector<ProtocolDecl *, 2> inheritedProtocols;
+          for (auto typeLoc : ED->getInherited()) {
+            if (!typeLoc.wasValidated() || typeLoc.isError() ||
+                typeLoc.isNull())
+              break;
+
+            // The type being inherited here can be a direct protocol
+            // conformance (e.g. extension Foo : Sequence), or an existential
+            // composition of conformances (e.g. extension Foo : Codable, where
+            // typealias Codable = Encodable & Decodable).
+            auto canType = typeLoc.getType()->getCanonicalType();
+            if (canType->isExistentialType()) {
+              auto layout = canType->getExistentialLayout();
+              for (auto protocolType : layout.getProtocols())
+                inheritedProtocols.push_back(protocolType->getDecl());
+            } else if (auto *inheritedNominal = canType->getAnyNominal()) {
```

Comment:
I don't think this 'else if' is reachable. isExistentialType() will already return true for ProtocolType.